src/main/java/com/google/cloud/discotoproto3converter/proto3/DocumentToProtoConverter.java

```diff
@@ -527,27 +532,51 @@ private Field schemaToField(Schema sch, boolean optional, String debugPreviousPa
     Message valueType = null;
     boolean repeated = false;
     Message keyType = null;
-    String debugCurrentPath =
-        debugPreviousPath + String.format("SCHEMA: %s\n%s\n----\n", name, description);
+    String debugCurrentPath = debugPreviousPath + String.format(".%s", name);
 
     if (trace) {
-      System.err.printf("*** schemaToField: \n%s", debugCurrentPath);
+      System.err.printf("*** schemaToField: %s\n", debugCurrentPath);
     }
 
     switch (sch.type()) {
       case ANY:
-        valueType = Message.PRIMITIVES.get("google.protobuf.Any");
+        switch (sch.format()) {
+          case VALUE:
+            valueType = Message.PRIMITIVES.get("google.protobuf.Value");
+            this.usesStructProto = true;
+            break;
+          case EMPTY:
+            // intentional fall-through
+          case ANY:
+            valueType = Message.PRIMITIVES.get("google.protobuf.Any");
+            break;
+          default:
+            throw new IllegalStateException(
+                String.format(
+                    "unexpected 'format' value ('%s') when processing ANY type in schema %s",
+                    sch.format().toString(), debugCurrentPath));
+        }
         break;
       case ARRAY:
-        repeated = true;
+        if (sch.format() == Format.LISTVALUE) {
+          valueType = Message.PRIMITIVES.get("google.protobuf.ListValue");
+          this.usesStructProto = true;
+          // the repeated semantics are inherent in the ListValue proto field type.
```

Comment:
Nit: perhaps beef up this comment for future readers - `google.protobuf.ListValue` is encoded as a `repeated` field in the [Protobuf JSON spec](https://protobuf.dev/programming-guides/proto3/#json), but the proto itself is still a message and not `repeated` as it appears in the `.proto` file, hence we do not mark `repeated = true` here.

long winded sorry haha

Reply:
Done. WDYT?

Reply:
Wonderful thank you!